src/sbd-common.c

```diff
@@ -118,6 +118,64 @@ usage(void)
                 , cmdname);
 }
 
+#define MAX_WATCHDOGS 64
+#define SYS_CLASS_WATCHDOG "/sys/class/watchdog"
+#define SYS_CHAR_DEV_DIR "/sys/dev/char"
+#define WATCHDOG_NODEDIR "/dev/"
+
+#define WATCHDOG_NODEDIR_LEN 5
+
+static bool
+is_watchdog(dev_t device)
+{
+    static int num_watchdog_devs = 0;
+    static dev_t watchdog_devs[MAX_WATCHDOGS];
+    struct dirent *entry;
+    int i;
+
+    /* populate on first call */
+    if (num_watchdog_devs == 0) {
+        DIR *dp;
+
+        watchdog_devs[0] = makedev(10,130);
+        num_watchdog_devs = 1;
+
+        /* get additional devices from /sys/class/watchdog */
+        dp = opendir(SYS_CLASS_WATCHDOG);
+        if (dp) {
+            while ((entry = readdir(dp))) {
+                if (entry->d_type == DT_LNK) {
+                    FILE *file;
+                    char entry_name[280];
```

Comment:
NAME_MAX + 6 would be better

(readdir() man page says "applications should avoid any dependence on the size of the d_name field" but also that POSIX specifies "at most NAME_MAX characters preceding the terminating null byte")

Alternatively you could use asprintf()

Reply:
not the only questionable array size ...
gcc gave me some unexpected complaints I didn't want to dig into before I could check if it looks as expected.

Reply:
Using NAME_MAX (+calculation) seems to satisfy gcc at least.
And in the end all we are risking (being written that NAME_MAX might not be enforced on some platforms) is truncation as everything writing to the strings is anyway length-guarded.